client/dashboard/me/profile-gravatar/edit-gravatar.tsx

```diff
@@ -83,77 +28,35 @@ const EditGravatar = ( { isEmailVerified = true, avatarUrl, userEmail }: EditGra
 
-	const handleUploadError = ( errorMessage: string ): void => {
-		// This could be replaced with a WordPress Notice component in the future
-		// eslint-disable-next-line no-console
-		console.error( errorMessage );
-	};
-
-	const uploadGravatarFile = async ( file: File ): Promise< void > => {
-		setIsUploading( true );
-		// crop file image to be a square
-		const croppedFile = await cropFile( file );
-		// Upload using wpcom REST API
-		wpcom.req
-			.post( {
-				method: 'POST',
-				path: '/gravatar-upload',
-				body: {},
-				apiNamespace: 'wpcom/v2',
-				formData: [
-					[ 'account', userEmail ],
-					[ 'filedata', croppedFile ],
-				],
-			} )
-			.then( () => {
-				setIsUploading( false );
-
-				// Create a temporary image preview
-				const fileReader = new FileReader();
-				fileReader.addEventListener( 'load', () => {
-					if ( typeof fileReader.result === 'string' ) {
-						setTempImage( fileReader.result );
-					}
-				} );
-				fileReader.readAsDataURL( file );
-			} )
-			.catch( () => {
-				setIsUploading( false );
-				handleUploadError(
-					__( 'Hmm, your new profile photo was not saved. Please try uploading again.' )
-				);
-			} );
-	};
-
-	const handleReceiveFile = ( event: React.ChangeEvent< HTMLInputElement > ): void => {
-		const files = event.target.files;
-		if ( ! files || ! files.length ) {
-			return;
-		}
-
-		const file = files[ 0 ];
-		const extension = path.extname( file.name ).toLowerCase().substring( 1 );
-
-		if ( ALLOWED_FILE_EXTENSIONS.indexOf( extension ) === -1 ) {
-			let errorMessage = '';
-
-			if ( extension ) {
-				// translators: %s is the file extension that is not supported
-				errorMessage = __(
-					'Sorry, %s files are not supported - please make sure your image is in JPG, GIF, or PNG format.'
-				).replace( '%s', extension );
-			} else {
-				errorMessage = __(
-					'Sorry, images of that filetype are not supported - please make sure your image is in JPG, GIF, or PNG format.'
-				);
-			}
-
-			handleUploadError( errorMessage );
-			return;
-		}
-
-		uploadGravatarFile( file );
-	};
+	// Initialize the Gravatar Quick Editor to manage avatars in a dedicated Gravatar UI
+	const quickEditorRef = useRef< GravatarQuickEditorCore | null >( null );
+
+	useEffect( () => {
+		quickEditorRef.current = new GravatarQuickEditorCore( {
+			email: userEmail,
+			scope: [ 'avatars' ],
+			utm: 'wpcomme',
+			onProfileUpdated: () => {
+				// Bust cache so the <img> reloads the latest avatar immediately
+				setTempImage( addQueryArgs( avatarUrl, { ver: Date.now() } ) as unknown as string );
+			},
+		} );
+
+		const onPageHide = () => {
+			try {
+				quickEditorRef.current?.close?.();
+			} catch ( _e ) {}
+		};
+		window.addEventListener( 'pagehide', onPageHide );
+
+		return () => {
+			window.removeEventListener( 'pagehide', onPageHide );
+			try {
+				quickEditorRef.current?.close?.();
+			} catch ( _e ) {}
+			quickEditorRef.current = null;
+		};
+	}, [ userEmail, avatarUrl ] );
```

Comment:
Do we need the `avatarUrl` here? I don't think we need to recreate the editor when the avatar image url changes.